src/store/actions/authaction.js

```diff
@@ -0,0 +1,33 @@
+const login = (user) => {
+  return { type: "LOGIN", payload: user };
+};
+
+const fail = (message) => {
+  return { type: "LOGIN_FAILURE", payload: message };
+};
+
+function success() {
+  return { type: "LOGIN_SUCCESS", payload: null };
+}
+
+function request() {
+  return { type: "REGISTER_REQUEST", payload: null };
+}
+
+export const middlewareLogin = (user) => {
+  return (dispatch) => {
+    // redux thunk will give you the dispatch
+    dispatch(request()); // loading is on, it will be more useful when you have backend later
+    try {
+      if (!user || !user.email || !user.password) {
+        // fail situation
+        dispatch(fail("you didnt give us email or password"));
+        return;
+      }
+      dispatch(login(user)); // success situation
```

Comment:
here if we really do want to simulate a backend, can you do a quick promise resolve in between to delay the login for ~2000ms? either do an await (and change to async or just)

https://stackoverflow.com/a/38956175/396324